packages/opentelemetry-core/src/common/attributes.ts

```diff
@@ -94,3 +124,46 @@ function isValidPrimitiveAttributeValue(val: unknown): boolean {
 
   return false;
 }
+
+function sanitizeHttpUrl(val: AttributeValue): AttributeValue | undefined {
+  let out: AttributeValue | undefined = undefined;
+
+  if (!isValidPrimitiveAttributeValue(val)) {
+    diag.warn(
+      `Invalid attribute value set for key: ${SemanticAttributes.HTTP_URL}. Unable to sanitize complex value.`
+    );
+
+    out = val;
+
+    return out;
+  }
```

Comment:
why do you need to check that if you only accept strings?

Reply:
the `URL` constructor only takes a string or other `URL` instances.